station-flattening-v0/src/main/java/net/modificationstation/stationapi/mixin/flattening/MixinLevelProperties.java

```diff
@@ -0,0 +1,40 @@
+package net.modificationstation.stationapi.mixin.flattening;
+
+import net.minecraft.level.LevelProperties;
+import net.minecraft.util.io.CompoundTag;
+import net.modificationstation.stationapi.api.registry.Identifier;
+import net.modificationstation.stationapi.impl.level.StationLevelProperties;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(LevelProperties.class)
+public class MixinLevelProperties implements StationLevelProperties {
+	@Unique private static final String DIMENSIONS_KEY = "StAPIDimensions";
+	@Unique private static CompoundTag dimensionsRoot;
```

Comment:
Is this supposed to be static?

Reply:
Yes, level properties are re-initiated again after player change dimension, if it will be not static we will get null (tested) with game crash